include/swift/AST/Decl.h

```diff
@@ -5694,6 +5694,25 @@ class AbstractFunctionDecl : public GenericContext, public ValueDecl {
 private:
   ParameterList *Params;
 
+// SWIFT_ENABLE_TENSORFLOW
+private:
+  /// The generation at which we last loaded derivative function configurations.
+  unsigned DerivativeFunctionConfigGeneration = 0;
+  /// Prepare to traverse the list of derivative function configurations.
+  void prepareDerivativeFunctionConfigurations();
+
+  /// A uniqued list of derivative function configurations.
```

Comment:
I think a note about who is responsible for populating this would be very helpful. It always takes me a while to figure out where these kinds of things get populated.

"Typechecking is responsible for populating derivative function configurations specified in the current module, and module loading is responsible for populating derivative function configurations specified in imported modules."

Reply:
Good idea! Done in https://github.com/apple/swift/pull/28608/commits/11c4db418e2a217a2392461e4f471a02750cd82b.